### DISCUSSION TITLE
Why does calculating 3D descriptors require the molecule to have a 3D conformer?

### DISCUSSION THREAD

**Discussion by fyy623** [rdkit/rdkit#7038] [FAQ]
I am confused here. If a molecule does not have a 3D conformer, can't we just "project" its 2D structure into a 3D space to make it a 3D conformer instead? Plus not having a 3D conformer should make it easier, in my opinion, to calculate a molecule's 3D properties, as it has simpler structure. So what is the purpose for setting such requirement after all?

![image](https://github.com/rdkit/rdkit/assets/59494795/30a9f3ad-1697-44be-afec-028e3800ebcc)

**Comment by greglandrum** [comment]
The 3D descriptors only make sense in the context of a 3D arrangement of atoms (i.e. a 3D conformer).
You can't calculate them at all if you don't have coordinates and it doesn't make sense to calculate them (i..e. they would not be meaningful) if those coordinates aren't 3D.

**Reply by fyy623** [reply]
I am still confused, I mean technically like I suggested above, we could project all 2D coordinates to 3D and calculate everything right?

Say we have a (2, 1) coordinate and all other coordiantes are in the same 2D space as well, can't we arbitrarily introduce a new axis such that we can convert (2, 1) into (2, 1, 0) and calculate it anyway, or this does not make sense mathematically/chemically?

**Reply by dehaenw** [reply]
you could do that (add z coordinate 0 to every atom) but the descriptors would be meaningless, like the plane of best fit for something that is forced to be on the (x,y)-plane. It should also be noted that what is called "2D descriptors" are actually not 2D in the sense they do not use or depend on 2D coordinates (which are usually optimized for display, not for geometric calculations) but rather use graph-based information like topological distance, atom invariants etc

**Reply by fyy623** [reply]
I see, then say I have a group of molecules, some of which have multiple conformers and some of which don't, then I can obviously calculate 3D descriptors of those with conformers. But for those don't, are there any make-up descriptors for them that can describe the same properties like those 3D descriptors do?

**Comment by thegodone** [comment]
If you project into 3D 2D you add zeros on the coordinates z. This is
absolutely mathematically correct but it does not represent 3D conformers
of a molecule. conclusion the 3D properties will be completely wrong
99.999% of the cases (except for linear molecules like C#C or C=O).

**Reply by fyy623** [reply]
Since mathematically it is just an extra axis that does not affect the actual calculation, why would it be chemically wrong? Can you provide an example?

**Reply by greglandrum** [reply]
@fyy623 if you want to understand the descriptors themselves and how they are calculated, I encourage you to read the references that are provided in the RDKit documentation (you have a screenshot above). Those articles tend to include the authors' thinking about why the descriptors were created.

**Comment by thegodone** [comment]
Hi, there is ways to generate conformers from smiles or molecules without conformers. Why not to do so ? 3D means we use the atomic distances in all codes so 3d without conformers (adding z 0.0 value) will produces wrong descriptors cause the distance is wrong and based on the image display that is not a real distance but a display distance. You need to make the difference on the rendering and the molecule structure object that have multiple 3D conformers. 3D descriptors must be done using accurate 3D conformers. If not the 3D descriptors are meaningless. To summarize you cannot do 3D descriptors without 3D distances aka conformers. It is like going back from grey to rgb for images… you will have not a rgb image but a grey one!Best Guillaume Envoyé de mon iPhoneLe 13 janv. 2024 à 07:39, Yuuya ***@***.***> a écrit :﻿
I see, then say I have a group of molecules, some of which have multiple conformers and some of which don't, then I can obviously calculate 3D descriptors of those with conformers. But for those don't, are there any make-up descriptors for them that can describe the same properties like those 3D descriptors do?

**Reply by fyy623** [reply]
I see, then i guess it is safe to assume that different conformers of the same molecule could generate different 3D descriptors? If so, what is the general strategy to summarize all of them together? does arithmetic mean of all conformers' descriptors make sense?

**Comment by thegodone** [comment]
Yes you ask the good question.

In general the conformers are relatively similar at a rotation
approximation.

We have true but minimal fluctuation in the 3D distance.

There were studies on the deviation of the descriptors based on
conformers, this is true but at the same time not very big.

We do not make average in RDkit but you can do it this is a good practice.
I would suggest using median instead to avoid outliers.

Best Guillaume

**Reply by fyy623** [reply]
This is really helpful, thank you!
